src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvokerCache.cs

```diff
@@ -85,7 +91,7 @@ public Entry GetCacheEntry(ControllerContext controllerContext)
             }
             else
             {
-                entry = new Entry(null, items, executor);
+                entry = new Entry(null, staticallyDefinedFilterItems, executor);
             }
```

Comment:
This logic, along with `allFiltersCached` needs to go away. We just want to store the items. This is a bit of a step back in perf, but we need to call the providers every request.